.github/workflows/test-build.yml

```diff
@@ -0,0 +1,23 @@
+# Based on sample workflow for building and deploying a Jekyll site to GitHub Pages
+name: Test Compile & Jekyll Build
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Comment:
Let's have this action drop the credentials after it's done by adding

```
        with:
            persist-credentials: false
```

See https://github.com/ossf/security-baseline/blob/main/.github/workflows/web-publish.yml#L27-L30

Reply:
```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
        with:
            persist-credentials: false
```

Like so?

Reply:
#nailedit